### ISSUE TITLE
Not finding linter and documentation listing wrong parameter

### ISSUE THREAD

**Issue by hisuwh** [glat/MultiLinter#9]
I'm trying to get this extension working with eslint and vue files. I have read through the other issue on this but my problem seems slightly different. 

First of all your documentation seems wrong.  It asks for your config to be something like this:

```json
{
    "eslint": {
      "enable": true,
      "fileExtensions": "js,vue"
    }
  }
```

But from looking at the debug output it seems it should be like this:

```diff
{
    "eslint": {
-     "enable": true,
+     "enabled": true,
      "fileExtensions": "js,vue"
    }
  }
```

Even with this configuration, it seems to not be able to find my local eslint.  This is the debug output:

```
Loading user defined tool configs...
Loading default tool configs from internal resources...
Default tool configs loaded from internal resources...
Searching config file in directory "C:\code\App\Application.WebApi\wwwroot\components\layout"
Searching config file in directory "C:\code\App\Application.WebApi\wwwroot\components"
Searching config file in directory "C:\code\App\Application.WebApi\wwwroot"
Searching config file in directory "C:\code\App\Application.WebApi"
Config file found in path "C:\code\App\Application.WebApi\.multilinterrc.json"
User defined tool configs loaded...


Formatter Name = prettier

            Enabled = false
         Executable = "prettier.cmd"
          Arguments = "--write "{filePath}""
     StdInArguments = "--stdin-filepath "{filePath}""
AdditionalArguments = ""
   InstallationType = "local"

Formatter Name = beautify

            Enabled = false
         Executable = "beautify.cmd"
          Arguments = "--file "{filePath}" --replace"
     StdInArguments = ""
AdditionalArguments = ""
   InstallationType = "local"

Linter Name = eslint

            Enabled = true
         Executable = "eslint.cmd"
          Arguments = ""{filePath}" -f json"
     StdInArguments = "--stdin --stdin-filename="{filePath}" -f json"
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = "js,vue"
        RuleHelpUrl = "https://eslint.org/docs/rules/{ruleId}"
    ResultsSelector = "$[0].messages"
     SourceSelector = "source"
       LineSelector = "line"
     ColumnSelector = "column"
    EndLineSelector = "endLine"
  EndColumnSelector = "endColumn"
    IsFatalSelector = "isFatal"
    MessageSelector = "message"
     RuleIdSelector = "ruleId"
RuleHelpUrlSelector = ""
   SeveritySelector = "severity"

Linter Name = stylelint

            Enabled = false
         Executable = "stylelint.cmd"
          Arguments = ""{filePath}" -f json"
     StdInArguments = "--stdin-filename="{filePath}" -f json"
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = ""
        RuleHelpUrl = "https://stylelint.io/user-guide/rules/{ruleId}"
    ResultsSelector = "$[0].warnings"
     SourceSelector = ""
       LineSelector = "line"
     ColumnSelector = "column"
    EndLineSelector = ""
  EndColumnSelector = ""
    IsFatalSelector = ""
    MessageSelector = "text"
     RuleIdSelector = "rule"
RuleHelpUrlSelector = ""
   SeveritySelector = "severity"

Linter Name = csslint

            Enabled = false
         Executable = "csslint.cmd"
          Arguments = ""{filePath}" --format=json"
     StdInArguments = ""
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = ""
        RuleHelpUrl = ""
    ResultsSelector = "$.messages"
     SourceSelector = "evidence"
       LineSelector = "line"
     ColumnSelector = "col"
    EndLineSelector = ""
  EndColumnSelector = ""
    IsFatalSelector = ""
    MessageSelector = "message"
     RuleIdSelector = "rule.id"
RuleHelpUrlSelector = "rule.url"
   SeveritySelector = "type"

Linter Name = jslint

            Enabled = false
         Executable = "jslint.cmd"
          Arguments = ""{filePath}" --json"
     StdInArguments = ""
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = ""
        RuleHelpUrl = ""
    ResultsSelector = "$[1]"
     SourceSelector = "evidence"
       LineSelector = "line"
     ColumnSelector = "character"
    EndLineSelector = ""
  EndColumnSelector = ""
    IsFatalSelector = ""
    MessageSelector = "reason"
     RuleIdSelector = "id"
RuleHelpUrlSelector = ""
   SeveritySelector = "id"

Linter Name = jshint

            Enabled = false
         Executable = "jshint.cmd"
          Arguments = ""{filePath}" --reporter "%APPDATA%\npm\node_modules\jshint-json\json.js""
     StdInArguments = ""
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = ""
        RuleHelpUrl = ""
    ResultsSelector = "$.data[0].functions"
     SourceSelector = ""
       LineSelector = "line"
     ColumnSelector = "character"
    EndLineSelector = "last"
  EndColumnSelector = "lastcharacter"
    IsFatalSelector = ""
    MessageSelector = ""
     RuleIdSelector = ""
RuleHelpUrlSelector = ""
   SeveritySelector = ""

Linter Name = sass-lint

            Enabled = false
         Executable = "sass-lint.cmd"
          Arguments = ""{filePath}" -v -f json"
     StdInArguments = ""
AdditionalArguments = ""
   InstallationType = "local"
     FileExtensions = ""
        RuleHelpUrl = ""
    ResultsSelector = "$[0].messages"
     SourceSelector = ""
       LineSelector = "line"
     ColumnSelector = "column"
    EndLineSelector = ""
  EndColumnSelector = ""
    IsFatalSelector = ""
    MessageSelector = "message"
     RuleIdSelector = "ruleId"
RuleHelpUrlSelector = ""
   SeveritySelector = "severity"

Searching linters to be used for file extension ".vue"
Linters found to be used for file extension ".vue" found: eslint,
Searching linter eslint in local mode
Searching linter eslint in directory "C:\code\App\Application.WebApi\wwwroot\components\layout"
Searching linter eslint in directory "C:\code\App\Application.WebApi\wwwroot\components"
Searching linter eslint in directory "C:\code\App\Application.WebApi\wwwroot"
Searching linter eslint in directory "C:\code\App\Application.WebApi"
Searching linter eslint in directory "C:\code\App"
Searching linter eslint in directory "C:\code"
Searching linter eslint in directory "C:\"
Linter eslint not found. Check your config.
```

I have eslint installed as a dependency under `"C:\code\App\Application.WebApi"`.  Though it does not seem to be in the `\node_modules\.bin` folder

**Comment by hisuwh** [comment]
I've managed to make this work for now using the global installation

**Comment by glat** [comment]
Multilinter checks for any path appending "\node_modules\.bin" starting from the file directory up to the root.
I never used vue, so I've to check where it is placed to modify searching of the eslint linter.

**Comment by glat** [comment]
Can you summarize your project directory tree?

**Comment by hisuwh** [comment]
> Can you summarize your project directory tree? 

Its in the comment above.  As mentioned though I'm not sure the problem is to do with multilinter as eslint did not seem to be present in the bin directory

**Comment by glat** [comment]
Yes it is but I'd like to understand where is placed the package.json and the node_modules directory. If there's no path ending with \node_modules\.bin\eslint.cmd MultiLinter can't find it.
Open a cmd, change current directory where package.json is placed and post the output of npm list --depth=0

**Comment by hisuwh** [comment]
package.json and node_modules are under "C:\code\App\Application.WebApi"

**Comment by glat** [comment]
Can you find the \node_modules\.bin\eslint.cmd in it?

**Comment by hisuwh** [comment]
```
hing-simmons@mycomputer:/mnt/c/code/App/App.Application.WebApi$ npm list --depth=0
asp.net@1.0.0 /mnt/c/code/App/App.Application.WebApi
├── @fortawesome/fontawesome-free@5.14.0
├── @fullcalendar/daygrid@5.1.0
├── @fullcalendar/timegrid@5.1.0
├── @fullcalendar/vue@5.1.0
├── @types/requirejs@2.1.28
├── UNMET PEER DEPENDENCY ace-builds@^1.2.2
├── ajv@6.12.3
├── ajv-keywords@3.5.1
├── babel-core@6.26.0
├── babel-loader@7.1.5
├── babel-plugin-jsx-v-model@2.0.3
├── babel-plugin-syntax-dynamic-import@6.18.0
├── babel-plugin-syntax-jsx@6.18.0
├── babel-plugin-transform-object-rest-spread@6.26.0
├── babel-plugin-transform-vue-jsx@3.7.0
├── babel-preset-env@1.6.0
├── UNMET PEER DEPENDENCY bootstrap@4.1.3
├── bootstrap-vue@2.15.0
├── chart.js@2.9.3
├── clean-webpack-plugin@0.1.19
├── compression-webpack-plugin@1.1.11
├── cross-env@7.0.2
├── css-loader@0.28.7
├── eks-upload@0.1.0
├── element-ui@2.3.8
├── UNMET PEER DEPENDENCY eslint@7.5.0
├── eslint-plugin-import@2.22.0
├── eslint-plugin-jsdoc@28.7.0
├── eslint-plugin-vue@6.2.2
├── file-loader@1.1.11
├── fuse.js@3.2.0
├── jquery@3.5.0
├── mini-css-extract-plugin@0.4.1
├── moment@2.22.2
├── node-sass@4.13.1
├── numbro@1.11.0
├── optimize-css-assets-webpack-plugin@3.2.0
├── pdfmake@0.1.39
├── popper.js@1.14.4
├── sass-loader@8.0.2
├── style-loader@0.19.0
├── survey-creator@1.7.20
├── survey-vue@1.7.20
├── surveyjs-widgets@1.7.20
├── tree-component@5.6.1
├── tree-vue-component@5.6.1
├── uglifyjs-webpack-plugin@1.2.7
├── url-loader@0.6.2
├── v-tooltip@2.0.3
├── vee-validate@2.0.3
├── UNMET PEER DEPENDENCY vue@2.5.18
├── vue-chartjs@3.5.0
├── vue-jstree@2.1.6
├── vue-loader@15.2.6
├── vue-momentjs@0.1.1
├── vue-multiselect@2.1.3
├── vue-nav-tabs@0.5.7
├── vue-password@1.1.1
├── vue-resource@1.3.5
├── vue-router@3.0.1
├── vue-search-select@2.5.0
├── vue-smooth-dnd@0.2.5
├── vue-tables-2@1.4.50
├── vue-template-compiler@2.5.18
├── vue-the-mask@0.11.1
├── vue2-autocomplete-js@0.2.2
├── vue2-filters@0.3.0
├── vuedraggable@2.24.0
├── vuejs-datepicker@0.9.25
├── vuex@3.0.1
├── webpack@4.41.6
├── webpack-bundle-analyzer@2.13.1
├── webpack-cli@3.3.12
└── webpack-merge@4.1.3
```

Maybe its something to do with me using WSL

**Comment by glat** [comment]
So, you installed packages by running npm install on WSL? I don't know if nodejs on Linux creates a "eslint.cmd" because it's a windows batch script. Can you check it? I can install WSL myself but It will take some time.

**Comment by glat** [comment]
I checked by installing a VM. Linux nodejs does not create a eslint.cmd, of course. So, you should try to delete node_nodules directory and run npm install on windows to create the *.cmd batch files used on Windows.

**Comment by hisuwh** [comment]
ok. good to know.  Thanks

**Comment by glat** [comment]
Can i close the issue?

**Comment by hisuwh** [comment]
yes
